docs/misc/file-route-conventions.md

```diff
@@ -290,6 +290,8 @@ If you want one of the special characters used for these route conventions to ac
 | `app/routes/weird-url.[_index].tsx` | `/weird-url/_index` |
 | `app/routes/dolla-bills-[$].tsx`    | `/dolla-bills-$`    |
 | `app/routes/[[so-weird]].tsx`       | `/[so-weird]`       |
+| `app/routes/reports.$id[.pdf].ts    | `/reports/123.pdf   |
+| `app/routes/reports.$id[.].ts       | `/reports/123.pdf   |
```

Comment:
Can we use skip the dynamic param in this example?  Params are supposed to be required to be the full segment, so `path: "reports/:id.pdf"` should _technically_ produce `params = { "id.pdf": "123" }` but it looks like due to a bug in our regex matching the `.` is not captured as part of the param name and you end up with `params = { "id": "123" }` and the partial param name works (accidentally) for this specific case.  That would probably be a breaking bug fix so we'll have to figure out how to best handle it in the future but for now let's not provide examples that call attention to it?

```suggestion
| `app/routes/$id.report[.pdf].ts    | `/123/report.pdf   |
```